app/src/main/java/com/sopt/now/presentation/auth/signup/SignupActivity.kt

```diff
@@ -0,0 +1,65 @@
+package com.sopt.now.presentation.auth.signup
+
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import com.sopt.now.utils.Constants.Companion.MAX_ID_LENGTH
+import com.sopt.now.utils.Constants.Companion.MIN_ID_LENGTH
+import com.sopt.now.utils.Constants.Companion.MIN_PASSWORD_LENGTH
+import com.sopt.now.R
+import com.sopt.now.databinding.ActivitySignupBinding
+import com.sopt.now.model.signup.RequestSignUpDto
+
+class SignupActivity : AppCompatActivity() {
+    private lateinit var binding: ActivitySignupBinding
+    private val signUpViewModel by viewModels<SignUpViewModel>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivitySignupBinding.inflate(layoutInflater).apply { setContentView(root) }
+        binding.btnSignup.setOnClickListener { signUp() }
+        observeSignUp()
+    }
+
+    private fun signUp() = signUpViewModel.signUp(getSignUpRequestDto())
+
+    private fun observeSignUp() {
+        signUpViewModel.status.observe(this) {
+            if (it) {
+                showToast((R.string.success_signup).toString())
+                backToLogin()
+            } else {
+                showToast(signUpViewModel.getErrorMessage()!!)
```

Comment:
```suggestion
                showToast(signUpViewModel.getErrorMessage()?:"알 수 없는 오류..")
```

!!타입은 최대한 피하는게 좋습니다! 여기선 이런식으로 바꾸어도 되는데 getErrorMessgae함수에서 null인 경우에대한 핸들링은 먼저 하는게 더 나을 수 도있을 것 같아요.